packages/orama/src/components/algorithms.ts

```diff
@@ -80,9 +89,14 @@ export function prioritizeTokenScores(
     lastTokenWithAllKeywords = 0
   }
 
+  const resultsWithIdAndScore: [number, number][] = [];
+  for (let i = 0; i < keywordsPerToken.length; i++) {
+    resultsWithIdAndScore.push([keywordsPerToken[i][0], keywordsPerToken[i][1]]);
+  }
+
```

Comment:
```suggestion
  const resultsWithIdAndScore: [number, number][] = new Array(keywordsPerToken.length);
  for (let i = 0; i < keywordsPerToken.length; i++) {
    resultsWithIdAndScore[i] = [keywordsPerToken[i][0], keywordsPerToken[i][1]];
  }
```

suggestion for better performance